### PR TITLE
Idefics2.

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by Narsil** [comment]
Mistral is gated and failing now :(

File: router/src/config.rs [hunk 1]
```diff
@@ -57,19 +57,46 @@ fn select_best_resolution(
...
+) -> (usize, usize) {
+    let current_height = npatches * num_patch_height;
+    let current_width = npatches * num_patch_width;
+
+    let aspect_ratio: f64 = width as f64 / height as f64;
+    let current_aspect_ratio: f64 = current_width as f64 / current_height as f64;
+    let (current_height, current_width) = if aspect_ratio > current_aspect_ratio {
+        let new_height = (height * current_width) / width;
+        // let padding = (current_height - new_height) / 2;
+        // let new_height = current_height - 2 * padding;
```
Comment (Narsil) [inline router/src/config.rs:75]:
```suggestion
```

File: router/src/config.rs [hunk 2]
```diff
@@ -57,19 +57,46 @@ fn select_best_resolution(
...
+    (unpadded_features, newline_features)
+}
+
 impl LlavaNext {
     pub fn get_number_of_features(&self, height: usize, width: usize) -> usize {
         let image_size = self.vision_config.image_size;
         let patch_size = self.vision_config.patch_size;
         assert!(image_size % patch_size == 0);
         let npatches = image_size / patch_size;
+        println!("{npatches} {image_size} {patch_size}");
```
Comment (Narsil) [inline router/src/config.rs:94]:
```suggestion
```

File: server/text_generation_server/models/custom_modeling/llava_next.py [hunk 3]
```diff
@@ -253,6 +229,10 @@ def forward(
                         self.config.image_grid_pinpoints,
                         self.config.vision_config.image_size,
                     )
+                    if image_sizes[image_idx][0].item() == 22:
+                        import ipdb
+
+                        ipdb.set_trace()
```
Comment (Narsil) [inline server/text_generation_server/models/custom_modeling/llava_next.py:235]:
```suggestion
```

**Review by OlivierDehaene — APPROVED** [review]
Nice

**Comment by RonanKMcGovern** [comment]
Hi @Narsil , the docs link [here](https://moon-ci-docs.huggingface.co/docs/text-generation-inference/main/en/HuggingFaceM4/idefics-9b-instruct) is failing.

It would be great to link somewhere to how you inference TGI with images for IDEFICS. Same @VictorSanh for the huggingface model card.

**Comment by VictorSanh** [comment]
@RonanKMcGovern one way you can pass multiple images is through the markdown syntax: ``.
i will update the model card

**Comment by RonanKMcGovern** [comment]
Ah thanks, I had guessed that but didn’t seem to get it working for two
images

I’ll try again.

**Pull request merged**